### PR TITLE
feature/DROPDOWN-Implement-dropdown

### PR DESCRIPTION
**Pull request by zanusilker**
- Implement dropdown select

### CODE REVIEW

File: src/components/dropdown.tsx [hunk 1]
```diff
@@ -0,0 +1,70 @@
+// outsource dependencies
+import React, { memo, useCallback, useState, useMemo, MouseEvent } from 'react';
+import { Button, MenuItem, ListItemIcon, ListItemText, Menu } from '@mui/material';
+import { SvgIconComponent, KeyboardArrowDown as KeyboardArrowDownIcon } from '@mui/icons-material';
+
+export interface DropdownOption {
+  key: string | number,
```
Comment (serejabogdan) [inline src/components/dropdown.tsx:7]: where is a three?
remove comma

Comment (serejabogdan) [inline src/components/dropdown.tsx:7]: where is a three?
remove comma

File: src/components/dropdown.tsx [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+  title: string
+  disabled?: boolean
+  options: Array<DropdownOption>
+}
+
+const Dropdown: React.FC<DropdownProps> = ({ title, disabled, options }) => {
+  const [ref, setRef] = useState<null | HTMLElement>(null);
+  const isOpen = Boolean(ref);
+  const handleClick = useCallback((event: MouseEvent<HTMLElement>) => setRef(event.currentTarget), []);
+  const handleClose = useCallback(() => setRef(null), []);
```
Comment (serejabogdan) [inline src/components/dropdown.tsx:23]: should you use useCallback here?

File: src/components/dropdown.tsx [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+interface DropdownProps {
+  title: string
+  disabled?: boolean
+  options: Array<DropdownOption>
+}
+
+const Dropdown: React.FC<DropdownProps> = ({ title, disabled, options }) => {
+  const [ref, setRef] = useState<null | HTMLElement>(null);
+  const isOpen = Boolean(ref);
+  const handleClick = useCallback((event: MouseEvent<HTMLElement>) => setRef(event.currentTarget), []);
```
Comment (serejabogdan) [inline src/components/dropdown.tsx:22]: [setRef]
Missing useCallback dependencies

**Review by serejabogdan — CHANGES_REQUESTED** [review]

File: src/components/dropdown.tsx [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+  handleClick: () => void
+}
+
+interface DropdownProps {
+  title: string
+  disabled?: boolean
+  options: Array<DropdownOption>
+}
+
+const Dropdown: React.FC<DropdownProps> = ({ title, disabled, options }) => {
```
Comment (alexandr-feschenko) [inline src/components/dropdown.tsx:19]:
```suggestion
const Dropdown: FC<DropdownProps> = ({ title, disabled, options }) => {
```

**Review by alexandr-feschenko — APPROVED** [review]

File: src/components/dropdown.tsx (hunk 3 above)
Comment (zanusilker) [reply src/components/dropdown.tsx:22]: No

**Pull request merged**